packages/core/upload/admin/src/utils/getFolderParents.js

```diff
@@ -0,0 +1,29 @@
+const buildRecursiveFolderParents = (results, node, searchedId) => {
+  let found = false;
+
+  if (node.value === parseInt(searchedId, 10)) {
+    found = true;
+  } else {
+    for (let i = 0; i < node.children.length && !found; i++) {
+      found = buildRecursiveFolderParents(results, node.children[i], searchedId);
+
+      if (found) {
+        results.push({ id: node.children[i].value, label: node.children[i].label });
+      }
+    }
+  }
+
+  return found;
+};
+
+const getFolderParents = (folderNode, folder) => {
+  const results = [];
+
+  if (folderNode && folder) {
+    buildRecursiveFolderParents(results, folderNode, folder);
+  }
+
+  return results.reverse();
+};
```

Comment:
What do you think of making this a bit easier: my suggestion would to be use `packages/core/upload/admin/src/components/SelectTree/utils/flattenTree.js` to flatten the tree first and then you can iterate more easily maybe:

```js
const getFolderParents = (folders, currentFolderId) => {
  const parents = [];
  const flatFolders = flattenTree(folders);
  const currentFolder = flatFolders.find(folder => folder.value === currentFolderId);

  let { parent } = currentFolder;

  while(parent !== undefined) {
    const parent = options.find(({ value }) => value === parent);
    parents.push(parent);
    parent = parent.parent;
  }
  
  return parents;
}
```

Reply:
I will test it, a little sad I didn't know about it earlier 🥲😅